src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,40 +1,66 @@
 import React from 'react';
 
+import { useAppSelector } from '../../app/hooks';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
+
 export const TodoFilter: React.FC = () => {
+  const filter = useAppSelector(state => state.filter);
+  const query = filter.query;
+  const dispatch = useDispatch();
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterSlice.actions.setStatus(event.target.value));
```

Comment:
The `event.target.value` is being dispatched directly without checking its type. Since the `setStatus` action probably expects a value of type `Status`, it would be better to ensure that the value is indeed of that type before dispatching the action. You can do this by creating a mapping from string to `Status` and use it to convert the value.